test: add toASCII and toUnicode punycode tests

claudiorodriguez opened the pull request:
<!--
Thank you for your pull request. Please review below requirements.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j8 test` (UNIX), or `vcbuild test nosign` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)
<!-- Provide affected core subsystem(s) (like doc, cluster, crypto, etc). -->
test

##### Description of change
<!-- Provide a description of the change below this comment. -->

- Add toUnicode and toASCII tests to test-punycode
- Refactor test-punycode.js to better organize test cases
- Change assert.equal to assert.strictEqual in test-punycode.js

claudiorodriguez commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/4944/

lpinca commented on test/parallel/test-punycode.js:
```diff
@@ -1,153 +1,237 @@
+  {
+    string: 'ليهمابتكلموشعربي؟',
+    encoded: 'egbpdaj6bu4bxfgehfvwxn',
+    decoded: '\u0644\u064A\u0647\u0645\u0627\u0628\u062A\u0643\u0644\u0645' +
```
Isn't `decoded` just a copy of `string`? If so can't we just use `string` and `encoded`?

silverwind replied:
I agree that we can probably just remove them if they server no purpose.

claudiorodriguez replied:
If that's how they appear in the RFC, I'd say maybe leave them just in case? It might also help in debugging, if the test ever starts failing, to have the bytecodes handy.

dongsik2026 commented on test/parallel/test-punycode.js:
```diff
@@ -1,153 +1,237 @@
-      '\uC5B4\uB97C\uC774\uD574\uD55C\uB2E4\uBA74\uC5BC\uB9C8\uB098\uC88B' +
-      '\uC744\uAE4C',
+  {
+    string: '세계의모든사람들이한국어를이해한다면얼마나좋을까',
```
I Guide this sentence like "세계의 모든 사람들이 한국어를 이해한다면 얼마나 좋을까"

claudiorodriguez commented:
I've removed the duplicitous `string` and left the `\u` encoded ones since they appear like that in the RFC, and also I've rewritten the `toUnicode` test to reflect its actual logic.

New CI: https://ci.nodejs.org/job/node-test-pull-request/4960/

claudiorodriguez commented:
CI Failure seems unrelated

claudiorodriguez commented:
I'll merge this tomorrow if there's no objections

claudiorodriguez commented:
Landed in c2f84877c5d186f6b503cb37981d1ae8cfee4f7c
